numpy/testing/nosetester.py

```diff
@@ -413,27 +401,62 @@ def test(self, label='fast', verbose=1, extra_argv=None,
             for warningtype in raise_warnings:
                 warnings.filterwarnings('error', category=warningtype)
             # Filter out annoying import messages.
-            warnings.filterwarnings('ignore', message='Not importing directory')
-            warnings.filterwarnings("ignore", message="numpy.dtype size changed")
-            warnings.filterwarnings("ignore", message="numpy.ufunc size changed")
-            warnings.filterwarnings("ignore", category=np.ModuleDeprecationWarning)
-            warnings.filterwarnings("ignore", category=FutureWarning)
+            sup.append(message='Not importing directory')
+            sup.append(message="numpy.dtype size changed")
+            sup.append(message="numpy.ufunc size changed")
+            sup.append(category=np.ModuleDeprecationWarning)
             # Filter out boolean '-' deprecation messages. This allows
             # older versions of scipy to test without a flood of messages.
-            warnings.filterwarnings("ignore", message=".*boolean negative.*")
-            warnings.filterwarnings("ignore", message=".*boolean subtract.*")
+            sup.append(message=".*boolean negative.*")
+            sup.append(message=".*boolean subtract.*")
+            # Filter out distutils cpu warnings (could be localized to
+            # distutils tests). ASV has problems with top level import,
+            # so fetch module for suppression here.
+            with warnings.catch_warnings():
+                warnings.simplefilter("always")
+                from ..distutils import cpuinfo
+            sup.append(category=UserWarning, module=cpuinfo)
+
             # Filter out some deprecation warnings inside nose 1.3.7 when run
             # on python 3.5b2. See
             #     https://github.com/nose-devs/nose/issues/929
+            # Note: it is hard to filter based on module for sup (lineno could
+            #       be implemented).
             warnings.filterwarnings("ignore", message=".*getargspec.*",
                                     category=DeprecationWarning,
                                     module="nose\.")
 
+            # Some of these warnings should likely be more spacific
+            # to a single module, etc.:
```

Comment:
Indeed, the warnings should only be suppressed when running the numpy tests, not tests of other packages. Otherwise it'll be much harder to fix those other packages up.